addons/stock/static/src/js/report_stock_reception.js

```diff
@@ -0,0 +1,214 @@
+        rr.on('click', '.o_report_reception_assign', this._onClickAssign.bind(this));
+        rr.on('click', '.o_report_reception_unassign', this._onClickUnassign.bind(this));
+        rr.on('click', '.o_report_reception_forecasted', this._onClickForecastReport.bind(this));
+        rr.on('click', '.o_print_label', this._onClickPrintLabel.bind(this));
+    },
+
+
+    _switchButton: function(button) {
+        if (button.innerText.includes('Unassign')) {
+            button.innerText = "Assign";
+            button.classList.add("o_report_reception_assign");
+            button.classList.remove("o_report_reception_unassign");
+        } else {
+            button.innerText = "Unassign";
+            button.classList.add("o_report_reception_unassign");
+            button.classList.remove("o_report_reception_assign");
+        }
+    },
+
+    /**
+     * Assign the specified move(s)
+     *
+     * @returns {Promise}
+     */
+    _onClickAssign: function(ev) {
+        let quantities = [];  // incoming qty amounts to assign
+        let moveIds = [];
+        let inIds = [];
+        let nodeToAssign = []
+        if (! ev.currentTarget.className.includes('o_assign_all')) {
+            nodeToAssign = [ev.currentTarget];
+            ev.target.closest('tbody').previousElementSibling.querySelectorAll('.o_print_label_all').forEach(button => button.removeAttribute('disabled'));
+        } else {
+            ev.target.style.display = 'none';
+            if (ev.target.name === "assign_all_link") {
+                this.iframe.contentDocument.querySelectorAll('.o_assign_all').forEach(b => b.style.display = 'none');
+                this.iframe.contentDocument.querySelectorAll('.o_print_label_all').forEach(button => button.removeAttribute('disabled'));
+                nodeToAssign = this.iframe.contentDocument.querySelectorAll('.o_report_reception_assign:not(.o_assign_all)');
+            } else {
+                // Local assign all
+                nodeToAssign = ev.target.closest('thead').nextElementSibling.querySelectorAll('.o_report_reception_assign:not(.o_assign_all)');
+                ev.target.closest('thead').nextElementSibling.querySelectorAll('.o_print_label_all').forEach(button => button.removeAttribute('disabled'));
+            }
+        }
+        nodeToAssign.forEach(node => {
+            node.closest('td').nextElementSibling.querySelectorAll('.o_print_label').forEach(button => button.removeAttribute('disabled'));
+            moveIds.push(parseInt(node.getAttribute('move-id')));
+            quantities.push(parseFloat(node.getAttribute('qty')));
+            inIds.push(JSON.parse(node.getAttribute('move-ins-ids')));
+            this._switchButton(node);
+        });
+
+        return this._rpc({
+            model: 'report.stock.report_reception',
+            args: [false, moveIds, quantities, inIds],
+            method: 'action_assign'
+        })
+    },
```

Comment:
For the `_onClickAssign` method, I suggest that:
* Save in a variable the `ev.currentTarget` and use it. The purpose of this is to use always the same element (instead of sometime `currentTarget`, sometime `target`), and make the code a little shorter (in my example, I named the variable `el` as it's a HTML element).
```javascript
const el = ev.currentTarget;
```
* For the first condition to know which button is clicked, I suggest you check the element's name (`'assign_link'`) instead of the absence of the class `o_assign_all` as you check the element's name.
It's shorter and it's easier to understand I think (about that, maybe add a comment for each condition like you did for the local assign, so we sure to know what part is about what button)
```javascript
if (el.name === 'assign_link') { // One line "Assign" button.
    ...
} else {
    el.style.display = 'none';
    if (el.name === "assign_all_link") { // Global "Assign All" button.
        ...
    } else { // Local "Assign All".
    ...
}
```
That said, it requires we change dynamically the button's name (as the name is defined in the XML template but then doesn't change, so the unassign button will still be named "assign_link" and the assign button will be named "unassign_link" if it was previously rended as the unassign button).
This can be done in the `_switchButton` method:
```javascript
if (button.name === 'assign_link') {
    button.name = 'unassign_link';
} else if (button.name === 'unassign_link') {
    button.name = 'assign_link';
}
```
* For the iframe's content, like the element target, you can save it into a variable to make the code a little shorter:
```javascript
const iframe = this.iframe.contentDocument;
iframe.querySelectorAll('.o_assign_all').forEach(button => button.style.display = 'none');
iframe.querySelectorAll('.o_print_label_all').forEach(button => button.removeAttribute('disabled'));
```
* I still don't like the chaining of `closest`, `nextElementSibling` and things like that, but I don't find an another way to do that without rework at least a little the XML template. And as we will have to redo the report later with the OWL framework, I think we can let that like this for now.

```suggestion
    _onClickAssign: function(ev) {
        const el = ev.currentTarget;
        const quantities = [];  // incoming qty amounts to assign
        const moveIds = [];
        const inIds = [];
        let nodeToAssign = [];
        if (el.name === 'assign_link') { // One line "Assign" button.
            nodeToAssign = [el];
            el.closest('tbody').previousElementSibling.querySelectorAll('.o_print_label_all').forEach(button => button.removeAttribute('disabled'));
        } else {
            el.style.display = 'none';
            if (el.name === "assign_all_link") { // Global "Assign All" button.
                const iframe = this.iframe.contentDocument;
                iframe.querySelectorAll('.o_assign_all').forEach(button => button.style.display = 'none');
                iframe.querySelectorAll('.o_print_label_all').forEach(button => button.removeAttribute('disabled'));
                nodeToAssign = iframe.querySelectorAll('.o_report_reception_assign:not(.o_assign_all)');
            } else { // Local "Assign All".
                nodeToAssign = el.closest('thead').nextElementSibling.querySelectorAll('.o_report_reception_assign:not(.o_assign_all)');
                el.closest('thead').nextElementSibling.querySelectorAll('.o_print_label_all').forEach(button => button.removeAttribute('disabled'));
            }
        }
        nodeToAssign.forEach(node => {
            node.closest('td').nextElementSibling.querySelectorAll('.o_print_label').forEach(button => button.removeAttribute('disabled'));
            moveIds.push(parseInt(node.getAttribute('move-id')));
            quantities.push(parseFloat(node.getAttribute('qty')));
            inIds.push(JSON.parse(node.getAttribute('move-ins-ids')));
            this._switchButton(node);
        });

        return this._rpc({
            model: 'report.stock.report_reception',
            args: [false, moveIds, quantities, inIds],
            method: 'action_assign'
        });
    },
```

Some of those suggestions can be applied to `_printLabel` and `_onClickUnassign` too